fixtures/pytest_store.py

```diff
@@ -207,3 +207,8 @@ def write_line(line, **kwargs):
         # resume capturing
         with diaper:
             store.capturemanager.resumecapture()
+
+
+def factory(cls, appliance_context=None, *args, **kwargs):
```

Comment:
I don't understand why we wouldn't use the "current_appliance" as that will automatically run the get_or_create if one isn't there yet